v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -449,103 +674,193 @@ github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
+│   │   │       └── "Scm"
+│   │   ├── NatGatewayState: *Enum (2 values)
+│   │   │   ├── "Disabled"
+│   │   │   └── "Enabled"
+│   │   ├── NotificationSenderEmail: *string
+│   │   ├── PublicIpAddressId: *string
+│   │   ├── PublicNetworkAccess: *Enum (2 values)
+│   │   │   ├── "Disabled"
+│   │   │   └── "Enabled"
+│   │   ├── PublisherEmail: *string
+│   │   ├── PublisherName: *string
+│   │   ├── Restore: *bool
+│   │   ├── VirtualNetworkConfiguration: *Object (1 property)
+│   │   │   └── SubnetResourceId: *string
+│   │   └── VirtualNetworkType: *Enum (3 values)
+│   │       ├── "External"
+│   │       ├── "Internal"
+│   │       └── "None"
+│   ├── Sku: *Object (2 properties)
+│   │   ├── Capacity: *int
+│   │   └── Name: *Enum (6 values)
+│   │       ├── "Basic"
+│   │       ├── "Consumption"
+│   │       ├── "Developer"
+│   │       ├── "Isolated"
+│   │       ├── "Premium"
+│   │       └── "Standard"
+│   ├── Tags: map[string]string
+│   └── Zones: string[]
+├── Service_Subscription_STATUS_ARM: Object (4 properties)
+│   ├── Id: *string
+│   ├── Name: *string
+│   ├── Properties: *Object (11 properties)
+│   │   ├── AllowTracing: *bool
+│   │   ├── CreatedDate: *string
+│   │   ├── DisplayName: *string
+│   │   ├── EndDate: *string
+│   │   ├── ExpirationDate: *string
+│   │   ├── NotificationDate: *string
+│   │   ├── OwnerId: *string
+│   │   ├── Scope: *string
+│   │   ├── StartDate: *string
+│   │   ├── State: *Enum (6 values)
+│   │   │   ├── "active"
+│   │   │   ├── "cancelled"
+│   │   │   ├── "expired"
+│   │   │   ├── "rejected"
+│   │   │   ├── "submitted"
+│   │   │   └── "suspended"
+│   │   └── StateComment: *string
+│   └── Type: *string
+├── Service_Subscription_Spec_ARM: Object (2 properties)
+│   ├── Name: string
+│   └── Properties: *Object (7 properties)
+│       ├── AllowTracing: *bool
+│       ├── DisplayName: *string
+│       ├── OwnerId: *string
+│       ├── PrimaryKey: *string
+│       ├── Scope: *string
+│       ├── SecondaryKey: *string
+│       └── State: *Enum (6 values)
+│           ├── "active"
+│           ├── "cancelled"
+│           ├── "expired"
+│           ├── "rejected"
+│           ├── "submitted"
+│           └── "suspended"
+└── Subscription: Resource
+    ├── Owner: Service
+    ├── Spec: Object (10 properties)
+    │   ├── AllowTracing: *bool
+    │   ├── AzureName: Validated<string> (2 rules)
+    │   │   ├── Rule 0: MaxLength: 256
+    │   │   └── Rule 1: Pattern: "^[^*#&+:<>?]+$"
+    │   ├── DisplayName: Validated<*string> (2 rules)
+    │   │   ├── Rule 0: MaxLength: 100
+    │   │   └── Rule 1: MinLength: 1
+    │   ├── OperatorSpec: *Object (1 property)
+    │   │   └── Secrets: *Object (2 properties)
+    │   │       ├── PrimaryKey: *genruntime.SecretDestination
+    │   │       └── SecondaryKey: *genruntime.SecretDestination
+    │   ├── Owner: *genruntime.KnownResourceReference
+    │   ├── OwnerId: *string
+    │   ├── PrimaryKey: Validated<*string> (2 rules)
```

Comment:
I've tried

```yaml
      Service_Subscription:
        PrimaryKey:
          $importConfigMapMode: optional
        SecondaryKey:
          $importConfigMapMode: optional
        $exportAs: Subscription
        $supportedFrom: v2.4.0
        $azureGeneratedSecrets:
          - PrimaryKey
          - SecondaryKey
```

and lots of other things and cant - I've spent quite a few hours on this and need help

Reply:
Actually, you're right on this - Primary and Secondary Keys should be `$azureGeneratedSecrets`. However we still need to mark them as secret since users can provide their own primary and secondary keys as per documentation - **Primary subscription key. If not specified during request key will be generated automatically.**

Reply:
@super-harsh, I've raised #3379 so that I can work on that separately. Is that a good way to move forward?

Reply:
Pro tip - add `--debug apimanagement` to the commandline when you run the generator; it will dump the current object model after _each stage of the pipeline_. The format used is similar to the `structure.txt` files, but without inlining things to the same degree.

Using that, I was able to establish that what you need to add is this:

``` yaml
      SubscriptionCreateParameterProperties:
        PrimaryKey:
          $isSecret: true
        SecondaryKey: 
          $isSecret: true
```

This gets us 90% of the way to where we need to be. The properties can be populated from secrets on the spec and sent to Azure, we have infrastructure to populate the secrets if they're generated by Azure, BUT `PrimaryKey` and `SecondaryKey` are elided from the Status, making it difficult to read the values. Not sure on the best approach just yet.

Reply:
The `--debug` flag is documented, but easy to miss. I added it in self-defense when I had trouble working out what to put in the config. 😁